pkg/mutation/assign_mutator.go

```diff
@@ -169,6 +204,22 @@ func IsValidAssign(assign *mutationsv1alpha1.Assign) error {
 	if err != nil {
 		return err
 	}
+	m, err := MutatorForAssign(assign)
+	if err != nil {
+		return err
+	}
+	pts, err := m.gatherPathTests()
```

Comment:
Between `IsValidAssign`, `ValidatePathTests`, and `getTester` - we end up calling `gatherPathTests` and `tester.New` twice. Is it worth refactoring a bit to avoid it?

Reply:
We could add some memoization, but I'm not sure there's much practical benefit.

In practice, I think the validation code gets called at most twice per mutator change -- once from the webhook and once for the controller. I'd expect the most amount of pathtests we'll see on any mutator is 2, maybe 3, so I was leaning toward keeping things simple and seeing if it became expensive later.

I'm more worried about the memoization cache not being copied as part of deepcopy -- that could add ongoing overhead for each mutation (I haven't double-checked to see whether a Mutate() call leads to a mutator being deep-copied).

Reply:
I could simply not call `getTester` here, but I think there are two advantages to doing so:

1) We don't need to worry about getTester adding other ways of breaking over time
2) We can more easily force memoization of the tester to happen during reconciliation, where latency is less impactful, instead of at webhook time.

Reply:
Fair enough 👍🏻